src/connections.jl

```diff
@@ -68,7 +71,8 @@ mutable struct Connection
             PQTypeMap(type_map),
             PQConversions(conversions),
             Atomic{Bool}(closed),
-            Mutex(),
+            Semaphore(1),
```

Comment:
Why would you want a `Semaphore(1)` rather than a Mutex?
Is the answer because we don't want reentrant behavour?
So that each lock can only be aquired once?

If so the `SpinLock()` is also not-reentrant.

Reply:
The `Semaphore` gets to use Conditions and the event loop, while a `SpinLock` could leave multiple threads spinning at once, eating CPU.

Reply:
orly?
Feels like a failing in julia 1.2 that that the same goodness is not extended to the LLVM atomic based locks.